app/components/heatmap.js

```diff
@@ -42,8 +42,9 @@ let initialRenderHeatmap = true;
 export default Component.extend({
   api: service(),
   timeZone: '',
-  toTimeZone(time, zone) {
-    return moment(time, TIME_FORMAT).tz(zone).format(TIME_FORMAT);
+  toEndDateTimeZone(time, zone) {
+    return moment(time, TIME_FORMAT).tz(zone).endOf('day').utc().
+      format(TIME_FORMAT);
   },
   buildFilterLabel: BUILDS_FILTER_LABELS['all'],
   buildYear: new Date().getFullYear(),
```

Comment:
The user timezone was ignored almost in every place where you re working with dates